### PR TITLE
Port GC changes from CoreRT

### PR DESCRIPTION
**Pull request by Suchiman**
I'm unsure if this change from CoreRT should be taken or reverted in CoreRT
https://github.com/Suchiman/coreclr/commit/b0e81dbaea17452f37f0720374fbbf8d774b570a

### CODE REVIEW

File: src/gc/gc.cpp [hunk 1]
```diff
@@ -5052,6 +5055,14 @@ extern "C" uint64_t __rdtsc();
         // all buffer access times being reported as equal in access_time().
         return 0;
     }
+#elif defined(_TARGET_WASM_)
```
Comment (jkotas) [inline src/gc/gc.cpp]: It would be better to just put the dummy implementation into the `#else` instead.

**Comment by jkotas** [comment]
cc @Maoni0

File: src/gc/gc.cpp (hunk 1 above)
Comment (Suchiman) [reply src/gc/gc.cpp]: Shall i collapse `_TARGET_ARM_`, `_TARGET_ARM64_` and `_TARGET_WASM_` (that all have identical implementations, just different comments) into `#else`?

Comment (jkotas) [reply src/gc/gc.cpp]: Yes, I think so.

Comment (Maoni0) [reply src/gc/gc.cpp]: yeah please do

---
In reply to: [292639177](https://github.com/dotnet/coreclr/pull/25092#discussion_r292639177) [](ancestors = 292639177)

**Comment by Suchiman** [comment]
What about this commit (i haven't included it in this PR because it looks suspicious to me)? https://github.com/Suchiman/coreclr/commit/b0e81dbaea17452f37f0720374fbbf8d774b570a
Should i also port this to CoreCLR or revert in CoreRT?

**Comment by Maoni0** [comment]
yes, you can port that one over - strictly speaking it's not needed in coreclr (only needed in corert) but since it's a completely trivial cost to pay we should just unify the changes.

**Pull request merged**